src/Layers/xrRenderPC_R3/r3_shaders.cpp

```diff
@@ -688,9 +688,19 @@ HRESULT CRender::shader_compile(pcstr name, IReader* fs, pcstr pFunctionName, pc
         includer Includer;
         LPD3DBLOB pShaderBuf = NULL;
         LPD3DBLOB pErrorBuf = NULL;
-        _result = D3DCompile(fs->pointer(), fs->length(), "", defines, &Includer, pFunctionName, pTarget, Flags, 0,
-            &pShaderBuf, &pErrorBuf);
 
+        if (ClearSkyMode)
+        {
+            if (!HW.oldD3DCompile)
+            HW.LoadOldD3DCompile();
+           _result = HW.oldD3DCompile(fs->pointer(), fs->length(), "", defines, &Includer, pFunctionName, pTarget,
+               Flags, 0, &pShaderBuf, &pErrorBuf);
+        }
+        else
+        {
+            _result = D3DCompile(fs->pointer(), fs->length(), "", defines, &Includer, pFunctionName, pTarget,
+                Flags, 0, &pShaderBuf, &pErrorBuf);
+        }
```

Comment:
Do it like this:
```suggestion
        _result = D3DCompile(fs->pointer(), fs->length(), "", defines, &Includer, pFunctionName, pTarget, Flags, 0,
            &pShaderBuf, &pErrorBuf);
            
        if (FAILED(_result) && ClearSkyMode)
        {
           _result = HW.OldD3DCompile(fs->pointer(), fs->length(), "", defines, &Includer, pFunctionName, pTarget,
               Flags, 0, &pShaderBuf, &pErrorBuf);
           if (SUCCEEDED(_result))
           {
               // TODO: Add log about errors and successful fallback to the old compiler
           }
        }
```
And don't forget to add log about errors and successful fallback to the old compiler =)